File: pdlearning/src/microservices/course-microservice/Api/Application/RequestDtos/RegistrationRequest/SearchRegistrationRequest.cs
using System;
using Microservice.Course.Application.Enums;
using Thunder.Platform.Core.Application.Dtos;

namespace Microservice.Course.Application.RequestDtos
{
    public class SearchRegistrationRequest : PagedResultRequestDto
    {
        public Guid? CourseId { get; set; }

        public Guid? ClassRunId { get; set; }

        public Guid? ExcludeAssignedAssignmentId { get; set; }

        public SearchRegistrationType SearchType { get; set; }

        public string SearchText { get; set; }

        /// <summary>
        /// Whether search text is used to filter for course name or not.
        /// </summary>
        public bool ApplySearchTextForCourse { get; set; }

        public CommonFilter UserFilter { get; set; }

        public CommonFilter Filter { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/course/dtos/search-registration-request.ts
import { IFilter } from '@opal20/infrastructure';
import { SearchRegistrationsType } from '../models/search-registrations-type.model';

export interface ISearchRegistrationRequest {
  courseId: string;
  classRunId: string;
  excludeAssignedAssignmentId?: string;
  searchType: SearchRegistrationsType;
  searchText: string;
  userFilter: IFilter;
  filter: IFilter;
  skipCount: number | null;
  maxResultCount: number | null;
}
